src/main.cpp

```diff
@@ -5,49 +5,216 @@
 
-static constexpr const char *version = "v.0.4.1";
+static constexpr const char *version = "v1.0.0";
+
+static constexpr const char *iconPath = "icon.png";
+
+static constexpr const std::chrono::milliseconds refreshIntervalBackground = 167ms; // 6fps
+static constexpr const std::chrono::milliseconds refreshIntervalFocused = 33ms;		// 30fps
 
-int autoStart = 1;
+static constexpr const int mainWindowWidth = 350;
+static constexpr const int mainWindowHeight = 304;
+
+static constexpr const float bitsToGB = 1073741824;
+
+#pragma region Config
+#pragma region Default settings
+// Initialization
+bool autoStart = 1;
 int minimizeOnStart = 0;
-float initialRes = 1.0f;
-
-float minRes = 0.60;
-float maxRes = 5.0f;
-long dataPullDelayMs = 250;
-long resChangeDelayMs = 1400;
-float minCpuTimeThreshold = 1.0f;
-float resIncreaseMin = 0.03f;
-float resDecreaseMin = 0.09f;
-float resIncreaseScale = 0.60f;
-float resDecreaseScale = 0.9;
-float resIncreaseThreshold = 0.75;
-float resDecreaseThreshold = 0.85f;
-int dataAverageSamples = 16;
-int resetOnThreshold = 1;
-int alwaysReproject = 0;
-float vramTarget = 0.8;
-float vramLimit = 0.9;
-int vramMonitorEnabled = 1;
-int vramOnlyMode = 0;
-int preferReprojection = 0;
-int ignoreCpuTime = 0;
-std::set<std::string> disabledApps;
-
-// NVML stuff
+// General
+int resChangeDelayMs = 2000;
+int dataAverageSamples = 128;
+std::string disabledApps = "steam.app.620980"; // Beat Saber
+std::set<std::string> disabledAppsSet;
+// Resolution
+int initialRes = 100;
+int minRes = 70;
+int maxRes = 350;
+int resIncreaseThreshold = 80;
+int resDecreaseThreshold = 88;
+int resIncreaseMin = 3;
+int resDecreaseMin = 7;
+int resIncreaseScale = 60;
+int resDecreaseScale = 90;
+float minCpuTimeThreshold = 0.6f;
+bool resetOnThreshold = true;
+// Reprojection
+bool alwaysReproject = false;
+bool preferReprojection = false;
+bool ignoreCpuTime = false;
+// VRAM
+int vramTarget = 80;
+int vramLimit = 90;
+bool vramMonitorEnabled = true;
+bool vramOnlyMode = false;
+#pragma endregion
+
+// God help me with these method names
+std::string spaceStringToNewlineString(const std::string &val)
+{
+	std::string result;
+	std::stringstream ss(val);
+	std::string word;
+
+	while (ss >> word)
+	{
+		result += word + "\n";
+	}
+
+	return result;
+}
+
+std::set<std::string> newlineStringToSet(const std::string &val)
+{
+	std::set<std::string> set;
+	std::stringstream ss(val);
+	std::string word;
+
+   for (std::string line; std::getline(ss, line, '\n');)
+        set.insert(line);
+
+	return set;
+}
+
+std::string setToSpaceString(std::set<std::string> &valSet)
+{
+	std::string result = "";
+
+	for (std::string val : valSet)
+	{
+		result += (val + " ");
+	}
+
+	return result;
+}
+
+bool loadSettings()
+{
+	// Get ini file
+	CSimpleIniA ini;
+	SI_Error rc = ini.LoadFile("settings.ini");
+	if (rc < 0)
+		return false;
+
+	// Startup
+	autoStart = std::stoi(ini.GetValue("Startup", "autoStart", std::to_string(autoStart).c_str()));
```

Comment:
`std::stoi` might throw an exception if the conversion is not possible.

So you could do....

```c++
try {
  autoStart = ...
...
} catch(...) { return false; }
```

This way the function will return false if there is a parsing error. Could be better than an exception.

Reply:
good idea